compiler/rustc_session/src/config.rs

```diff
@@ -325,11 +329,26 @@ impl Default for TrimmedDefPaths {
 
 /// Use tree-based collections to cheaply get a deterministic `Hash` implementation.
 /// *Do not* switch `BTreeMap` out for an unsorted container type! That would break
-/// dependency tracking for command-line arguments.
-#[derive(Clone, Hash, Debug)]
+/// dependency tracking for command-line arguments. Also only hash keys, since tracking
+/// should only depend on the output types, not the paths they're written to.
+#[derive(Clone, Debug)]
 pub struct OutputTypes(BTreeMap<OutputType, Option<PathBuf>>);
 
-impl_stable_hash_via_hash!(OutputTypes);
```

Comment:
Does this just forward the HashStable impl to Hash? Theoretically it can stay then, right? Since Hash is still implemented, just not derived?

Reply:
Ah, yeah. I changed the stable hash first thinking it was what would be used for the dep hash, but I never saw it being called so I changed the Hash impl, but didn't think to revert this change. I'll flip it back.